### ISSUE TITLE
Plot insert size distributions stratified by capture platform

### ISSUE THREAD

**Issue by gtiao** [broadinstitute/gnomad_qc#145]

**Comment by jkgoodrich** [comment]
I don't see any insert size annotations for v2 exomes on the metadata HT.

![Screen Shot 2021-09-30 at 4.35.28 PM.png](https://images.zenhubusercontent.com/5f4d3fb5900ef2def7626452/d4e5c769-6d2d-4056-b11d-2d5be0f7084d)

**Comment by gtiao** [comment]
This analysis was requested in a gnomAD meeting discussion on v4 sample QC: https://docs.google.com/document/d/1XSYWoTlS_vXEoltKo-sUpwb-jsLGz5AL_I9XDLW0a-s/edit

**Comment by jkgoodrich** [comment]
Yeah, I don't have v2 median insert size for exomes so I wasn't able to check this. @klaricch said she thinks she only looked at insert size in genomes and the gnomad v2 sample QC hard filtering only used insert size for genomes https://github.com/broadinstitute/gnomad_qc/blob/f42560719d30285ec6a491c3ce405b4b49b68e6a/gnomad_qc/v2/sample_qc/apply_hard_filters.py#L56

**Comment by gtiao** [comment]
Reading over the notes from the discussion, it does appear that the proposal to analyze insert size with respect to capture platform is a new effort. Since we don't have this data on v2, let's try this on v4 (assuming it's available). This may help us to decide whether it's important to infer capture platform before starting the rest of the QC pipeline.

**Comment by jkgoodrich** [comment]
- Run on the v4 test dataset with Julia's test platforms (and v2 assigned platforms)
- Run on the full v2 dataset with v4 data

**Comment by jkgoodrich** [comment]
This notebook is in flux, but has how I did the new platform inference on the test dataset: gs://gnomad-julia/gnomad_v4/impute_sex_tests_clean.ipynb


I think the main part needed for this is:

```
ukb_meta_ht = hl.read_table("gs://gnomad-julia/gnomad_v4/synthetic_dataset/ukb.exomes.455K.synthetic_dataset_samples.ht")
gnomad_meta_ht = hl.read_table("gs://gnomad-julia/gnomad_v4/synthetic_dataset/gnomad.exomes.v4.synthetic_dataset_samples.ht")
gnomad_v3_meta_ht = hl.read_table("gs://gnomad-julia/gnomad_v4/synthetic_dataset/gnomad.genomes.v3.synthetic_dataset_samples.ht")
gnomad_meta_ht = gnomad_meta_ht.key_by(s=gnomad_meta_ht.seq_id)

gnomad_meta_sex_ht = gnomad_meta_ht.select(
    platform=hl.str(gnomad_meta_ht.v2_meta.qc_platform),
    f_stat = gnomad_meta_ht.v2_meta.f_stat,
    is_female = gnomad_meta_ht.v2_meta.is_female,
    ambiguous_sex = gnomad_meta_ht.v2_meta.ambiguous_sex,
    sex_aneuploidy = gnomad_meta_ht.v2_meta.sex_aneuploidy,
    sex = hl.case()
    .when(gnomad_meta_ht.v2_meta.ambiguous_sex, "ambiguous_sex")
    .when(gnomad_meta_ht.v2_meta.sex_aneuploidy, "sex_aneuploidy")
    .when(gnomad_meta_ht.v2_meta.is_female, "XX")
    .when(~gnomad_meta_ht.v2_meta.is_female, "XY")
    .or_missing(),
    in_ukb="other v4",
)
ukb_meta_sex_ht = ukb_meta_ht.select(
    platform="ukb",
    f_stat = ukb_meta_ht.sex_imputation.f_stat, 
    is_female = ukb_meta_ht.sex_imputation.is_female, 
    autosomal_mean_dp=ukb_meta_ht.sex_imputation.chr20_mean_dp, 
    x_mean_dp=ukb_meta_ht.sex_imputation.chrX_mean_dp, 
    y_mean_dp=ukb_meta_ht.sex_imputation.chrY_mean_dp, 
    x_ploidy=ukb_meta_ht.sex_imputation.chrX_ploidy, 
    y_ploidy=ukb_meta_ht.sex_imputation.chrY_ploidy, 
    ambiguous_sex = ukb_meta_ht.sample_filters.ambiguous_sex, 
    sex_aneuploidy = ukb_meta_ht.sample_filters.sex_aneuploidy, 
    sex = hl.case()
    .when(ukb_meta_ht.sex_imputation.sex_karyotype == "XX", "XX")
    .when(ukb_meta_ht.sex_imputation.sex_karyotype == "XY", "XY")
    .when(ukb_meta_ht.sample_filters.ambiguous_sex, "ambiguous_sex")
    .when(ukb_meta_ht.sample_filters.sex_aneuploidy, "sex_aneuploidy")
    .or_missing(),
    sex_karyotype = ukb_meta_ht.sex_imputation.sex_karyotype,
    in_ukb="UKBB"
)

v3_meta_sex_ht = gnomad_v3_meta_ht.select(
    f_stat = gnomad_v3_meta_ht.sex_imputation.impute_sex_stats.f_stat, 
    is_female = gnomad_v3_meta_ht.sex_imputation.is_female, 
    autosomal_mean_dp=gnomad_v3_meta_ht.sex_imputation.chr20_mean_dp, 
    x_mean_dp=gnomad_v3_meta_ht.sex_imputation.chrX_mean_dp, 
    y_mean_dp=gnomad_v3_meta_ht.sex_imputation.chrY_mean_dp, 
    x_ploidy=gnomad_v3_meta_ht.sex_imputation.chrX_ploidy, 
    y_ploidy=gnomad_v3_meta_ht.sex_imputation.chrY_ploidy, 
    ambiguous_sex = gnomad_v3_meta_ht.sample_filters.ambiguous_sex, 
    sex_aneuploidy = gnomad_v3_meta_ht.sample_filters.sex_aneuploidy, 
    sex = hl.case()
    .when(gnomad_v3_meta_ht.sex_imputation.sex_karyotype == "XX", "XX")
    .when(gnomad_v3_meta_ht.sex_imputation.sex_karyotype == "XY", "XY")
    .when(gnomad_v3_meta_ht.sample_filters.ambiguous_sex, "ambiguous_sex")
    .when(gnomad_v3_meta_ht.sample_filters.sex_aneuploidy, "sex_aneuploidy")
    .or_missing(),
    sex_karyotype = gnomad_v3_meta_ht.sex_imputation.sex_karyotype,
)

meta_ht = gnomad_meta_sex_ht.union(ukb_meta_sex_ht, unify=True)

platform_ht = hl.read_table("gs://gnomad-julia/gnomad_v4/test_coverage_platform_dp_0.ht")

platform_pcs_ht = platform_ht.select(
    platform=hl.str(platform_ht.qc_platform),
    **{f'PC{i}': platform_ht.scores[i-1] for i in range(1,11)},
    v2_platform=meta_ht[platform_ht.key].platform
)
```

Then you can use the `platform` and `v2_platform` annotations on `platform_pcs_ht`.

**Comment by jkgoodrich** [comment]
Oh, yes, and also the notebook where I attempted to do this with v2 and found that there was no insert size info for the exomes: gs://gnomad-julia/gnomad_v4/sample_qc_improvement_plots.ipynb

**Comment by klaricch** [comment]
plots are here:
https://the-tgg.slack.com/archives/C018PB68VU5/p1643747162513759

**Comment by gtiao** [comment]
As discussed in the March 1 project meeting, we will be retaining all samples with low (platform-specific) insert sizes during the sample QC process and doing a downstream analysis to see if this class of samples is enriched for other types of sample QC outliers. See ticket #430
